File: Queue/Queue.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Queue
{
    class Program
    {
        static void Main(string[] args)
        {
            Queue queue = new Queue();
            queue.Enqueue(4);
            queue.Enqueue(5);

            Console.WriteLine((int)queue.Dequeue());
            Console.WriteLine((int)queue.Dequeue());
        }
    }

    class Queue
    {
        LinkedList<Object> list;

        public Queue()
        {
            list = new LinkedList<Object>();
        }

        public void Enqueue(Object data)
        {
            list.AddFirst(data);
        }

        public Object Dequeue()
        {
            LinkedListNode<Object> node = list.First;
            list.RemoveFirst();
            return node.Value;
        }

        public Object Peek()
        {
            LinkedListNode<Object> node = list.First;
            list.RemoveFirst();
            return node.Value;
        }
    }

}

File: Queue/Queue.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */

package stackjava;

/**
 *
 * @author ashab
 */
public class Queue {
    
    LinkedListNode start;
    LinkedListNode end;
    
    public void Enqueue(Object data)
    {
        if (start == null)
        {
            start = new LinkedListNode(data);
            end = start;
        }
        else
        {        
            end.Next = new LinkedListNode(data);;
            end = end.Next;        
        }
    }
    
    public Object Dequeue()
    {
        if (start != null)
        {
            Object result = start.Data;
            start = start.Next;
            return result;
        }
        
        return null;
    }
    
    public Object Peek()
    {
        if (start != null)
        {
            Object result = start.Data;
            return result;
        }
        
        return null;        
    }
}
